packages/react/src/select/root/SelectRoot.tsx

```diff
@@ -17,6 +17,12 @@ import { stringifyAsValue } from '../../utils/resolveValueLabel';
  *
  * Documentation: [Base UI Select](https://base-ui.com/react/components/select)
  */
+export function SelectRoot<Value, Multiple extends boolean | undefined = false>(
+  props: SelectRootControlledProps<Value, Multiple>,
+): React.JSX.Element;
+export function SelectRoot<Value, Multiple extends boolean | undefined = false>(
+  props: SelectRootUncontrolledProps<Value, Multiple>,
+): React.JSX.Element;
```

Comment:
Wrappers also would need to create overloads to have correct inference for the consumers, but these types seem overly complex to specify on the namespace. Maybe they need to just recreate them locally.

Reply:
To be fair, Combobox has the same overloads already. The overloads seemed necessary to make the type inference work correctly in all the `spec` cases I made, maybe a skill issue on my behalf, or TS limitations. It's a tricky topic 🤔

Reply:
Ok.
Then maybe it could make sense to add documentation about this fact if majority of our users are going to wrap these components? 🤔

Reply:
I bumped my head into this while upgrading from beta.4 to beta.5. This pr is not in the breaking changes. And no docs about this. Might be confusing?

<img width="813" height="288" alt="SCR-20251118-gngv" src="https://github.com/user-attachments/assets/27bf09e1-c8ef-41fc-989e-9c3f1b9103de" />